### PR TITLE
fix error in cache when masked isSensitive fields

### LINKED ISSUE #578: Fix error in cache when masked isSensitive fields
Fix the value with isSensitive in the cache, and the value in the previous method cache will also be masked.

### PR DESCRIPTION
**Pull request by jinyangyang222**
fixed #578

### CODE REVIEW

**Review by swaroopar — APPROVED** [review]
LGTM

File: modules/credential/src/test/java/org/eclipse/xpanse/modules/credential/CredentialCenterTest.java [hunk 1]
```diff
@@ -258,7 +259,7 @@ void testGetCredentials_WithRequestedCredentialType() {
 
         credentialsStore.getCredential(csp, requestedCredentialType, xpanseUser);
 
-        assertEquals(Collections.singletonList(credential), result);
+        assertNotEquals(Collections.singletonList(credential), result);
```
Comment (swaroopar) [inline modules/credential/src/test/java/org/eclipse/xpanse/modules/credential/CredentialCenterTest.java:262]: If the list is not same, we can now add assertions to validate if the values inside the list are same.

**Review by swaroopar — CHANGES_REQUESTED** [review]
extend test assertions pls.

**Review by swaroopar — APPROVED** [review]
LGTM

File: modules/credential/src/test/java/org/eclipse/xpanse/modules/credential/CredentialCenterTest.java (hunk 1 above)
Comment (jinyangyang222) [reply modules/credential/src/test/java/org/eclipse/xpanse/modules/credential/CredentialCenterTest.java:262]: Okay, I added it.

**Pull request merged**